corehq/apps/translations/validator.py

```diff
@@ -158,10 +158,15 @@ def _compare_sheet(self, module_or_form_id, uploaded_rows, for_type):
         :return: diff generated by ghdiff or None
         """
         columns_to_compare = COLUMNS_TO_COMPARE[for_type] + self.lang_cols_to_compare
-        header_indices = {
-            column_name: self._get_current_header_index(module_or_form_id, column_name)
-            for column_name in columns_to_compare
-        }
+        try:
+            header_indices = {
+                column_name: self._get_current_header_index(module_or_form_id, column_name)
+                for column_name in columns_to_compare
+            }
+        except KeyError:
```

Comment:
This is a generic exception and generally I believe we should avoid doing that unless we are sure of the very specific case. So can we either move this exact check to a more appropriate place or may be update the check in a way that we catch only the case we need?

Reply:
Do you mean that the try-catch block should be inside `_get_current_header_index`?
The function just extracts fields from `self.current_headers[module_or_form_id]`, I felt that KeyError should suffice the need here.
Can you suggest some other Exception classes that can be used here?

Reply:
Ya it makes sense in `_get_current_header_index` where it can raise that exception or we can raise it here itself if `module_or_form_id` is not in `self.current_headers`. Although having it in `_get_current_header_index` will handle all places where it is called.

It's hard/not possible to say what all Exception classes should be used here, so we should cover exactly the case we care about.

Reply:
Updated the code and moved the try-catch block inside `_get_current_header_index` and since we were only looking for `KeyError` so kept the same `Exception` class.